### PR TITLE
Accelerate runner helper method

### PR DESCRIPTION
**Pull request by avishniakov**
### Key changes
- Adding `zenml.integrations.huggingface.steps.run_with_accelerate` to be able to run any step using Accelerate (obviously, the step should be written in a way that it makes sense to run it with Accelerate). This function is supported by a utility, which wraps any function into a Click CLI script (which is needed by most of the Distributed train tools). CLI is quite limited, so it accepts only `(str, int, float, bool, Path, tuple)` as params. This can be maybe extended in the future.
How it works?
```python
from zenml.integrations.huggingface.steps import run_with_accelerate

@pipeline
def llm_peft_full_finetune():
    ...
    ft_model_dir = run_with_accelerate(finetune)(
        base_model_id=base_model_id,
        dataset_dir=datasets_dir,
        load_in_8bit=load_in_8bit,
    )
    ...
```
Running the same step without accelerating would be:
```python
@pipeline
def llm_peft_full_finetune():
    ...
    ft_model_dir = finetune(
        base_model_id=base_model_id,
        dataset_dir=datasets_dir,
        load_in_8bit=load_in_8bit,
    )
    ...
```
- Helper method `cleanup_gpu_memory` to clean up GPU memory at the step start. It has a side effect on the environment, so there is a must `force` param to be passed to protect that.

Companion PR: https://github.com/zenml-io/zenml-projects/pull/102

Sample run: `runs/5080958e-31f5-46b0-8e0b-fbf7086f1ff4` in Demo

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by avishniakov** [comment]
@strickvl , I can see that the docs for the `finetuning-llms` are not quite ready. Shall I wait for you or how to proceed best to documents these changes?

**Comment by strickvl** [comment]
> @strickvl , I can see that the docs for the `finetuning-llms` are not quite ready. Shall I wait for you or how to proceed best to document these changes?

Could you maybe make changes directly on the `feature/gro-1047-docs` branch directly @avishniakov and I'll touch them up. You can see I added a short section at the bottom of `docs/book/how-to/training-with-gpus/training-with-gpus.md` page. Or just send something to me in Discord and I'll add it. I couldn't find/think of the good way to document this with code examples, but probably you're in a better place to do so.

File: src/zenml/utils/cuda_utils.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    Args:
+        force: whether to force the cleanup of GPU memory (must be passed explicitly)
+    """
+    if not force:
+        logger.warning(
+            "This will clean up all GPU memory on current physical machine. "
+            "This action is considered to be dangerous by default, since "
+            "it might affect other processes running in the same environment. "
+            "If this is intended, please, explicitly pass `force=True`."
```
Comment (strickvl) [inline src/zenml/utils/cuda_utils.py:37]:
```suggestion
            "If this is intended, please explicitly pass `force=True`."
```

**Review by strickvl — CHANGES_REQUESTED** [review]
Just one tiny comment otherwise looks good to me. Is there any testing we can add for the function utils? Also wondering if we should have an e2e test script which runs a simple (CPU-backed) pipeline in CI but using `run_with_accelerate`? I'm a bit worried by how quickly HF things can break, esp with something like `accelerate` so I'd like to get advance failure notice of API changes that way if possible.

**Review by strickvl — APPROVED** [review]
Thanks for adding the tests. This LGTM now, providing you get the tests passing (there's a security one failing atm e.g.)

**Comment by avishniakov** [comment]
@bcdurak, as discussed this one is ready to be merged, but can give more work to the pydantic merge. LMK, when can I merge it?

**Pull request merged**